model/generate.go

```diff
@@ -184,6 +185,22 @@ func updateParserProject(ctx context.Context, settings *evergreen.Settings, v *V
 		return nil
 	}
 
+	// If this is the first time the parser project has been updated by a generator, cache a copy of
+	// the parser project representing its state before any generators updated it.
+	if len(pp.UpdatedByGenerators) == 0 {
+		oldPP, err := ParserProjectFindOneByID(ctx, settings, v.ProjectStorageMethod, v.Id)
+		if err != nil {
+			return errors.Wrapf(err, "finding parser project '%s' from before task generation", v.Id)
+		}
+		if oldPP == nil {
+			return errors.Errorf("parser project '%s' not found", v.Id)
+		}
+		oldPP.Id = fmt.Sprintf("%s_%s", "pre_generation", oldPP.Id)
+		if _, err = ParserProjectUpsertOneWithS3Fallback(ctx, settings, evergreen.ProjectStorageMethodDB, oldPP); err != nil {
```

Comment:
This seems like it'll have bugs if the pre-generated is in a different location than the post-generated one. For example, if the pre-generated copy is in the DB, but the post-generated copy is in S3. Evergreen won't be able to figure out where you put the pre-generated parser project because [it finds the pre-generated parser project based on the post-generated storage method](https://github.com/evergreen-ci/evergreen/blob/d90b82f86c1c1d30dfea1324d2877d943df0615d/model/project.go#L1365).

One thought is that maybe you can set a separate, special field for the pre-generated parser project storage method and set it here. Alternatively, you can avoid the new field by redundantly storing the pre-generated parser project in S3 when the post-generated parser project is switched over to S3.

Reply:
Makes sense, I opted to add one more flag to the version document